src/parser.rs

```diff
@@ -612,7 +613,50 @@ fn key_value_pairs(i: &[u8]) -> IResult<&[u8], HashMap<String, String>> {
     )(i)
 }
 
-fn key_value_pair(i: &[u8]) -> IResult<&[u8], (String, String)> {
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum QuotedOrUnquoted {
+    Unquoted(String),
+    Quoted(String),
+}
+
+impl QuotedOrUnquoted {
+    pub fn value(self) -> String {
+        match self {
+            QuotedOrUnquoted::Unquoted(s) => s,
+            QuotedOrUnquoted::Quoted(u) => u,
+        }
+    }
+}
+
+impl Default for QuotedOrUnquoted {
+    fn default() -> Self {
+        QuotedOrUnquoted::Quoted(String::new())
+    }
+}
+
+impl From<&str> for QuotedOrUnquoted {
```

Comment:
Might also want to implement `AsRef<str>` and `Deref<Target=str>` here

Reply:
Why? It doesn't seem to be needed

Reply:
Makes it easier for users to deal with the `QuotedOrUnquoted` enum in the public API. They wouldn't have to match on the enum if they just want to get the string out of it.